### ISSUE TITLE
Added Glue lock via Homebridge to Homekit, worked fine for a couple of hours and now im getting this error message (see inside)

### ISSUE THREAD

**Issue by phattrance** [GlueHome/homebridge-plugin#26]
So i added my glue lock via the GlueHome Homebridge Plugin ( https://github.com/GlueHome/homebridge-plugin#readme ). I used the basic info (user and password, since glue doesnt share API anymore).  I added it to home kit and it was working for for a couple of hours, i could lock and unlock the door etc.
But now im getting this error message in homebridge status and im not able to use the glue lock via homekit anymore. Does anyone know what this message mail and howe i can fix this?  Ive restarted my mac mini, homebridge and router without any results. I even uninstalled and installed the plugin (didnt work)

Error message:
[10/25/2022, 3:12:27 PM] [@gluehome/homebridge-gluehome] Error authenticating: Error: Error: Request failed with status code 401
at Object.issueApiKey (/usr/local/lib/node_modules/@gluehome/homebridge-gluehome/src/api/client.ts:27:11)
at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by andycb** [comment]
Hey, I can look the logs on our send to see what's happening but I'll need to find your Glue account. Can send a message to `support@gluehome.com` and let us know the email address you signed up with.

**Comment by phattrance** [comment]
Hi andy, i just send you an email with my email adress! 

Thanks for checking this for me.

**Comment by andycb** [comment]
Thanks. Got it and was able to pull up your account details. 

It seems that the above error was thrown while trying to generate an API key to use, but it was unable to validate your email / password. You mentioned you updated your email while debugging the issue, so that may be what caused this specific error (if not the underlying one)?

You have 9 API keys on your account, all generated today, so it looks like bridge is not storing the API key for some reason and is trying again on every poll. (By the way, we have a limit of 10, so if you want me to reset these let me know).

Can you verify what you have entered in your homebridge config.json file, please? if you have only your API key and not your original password, it should prevent it from attempting to constantly regenerate they key.

It should look something like this:
<img width="433" alt="image" src="https://user-images.githubusercontent.com/4417762/197830366-1e2819dd-e692-47f7-a240-ccf3da112ef1.png">

**Comment by phattrance** [comment]
Hi,

Thank you for the quick reply,

When I installed the glue plugin, I started with creating API connection, but every CURL gave me 401 errors, so I tried different ways to create them, and all of the were unsuccessful, so I basically gave up and went with the ”username and password”, so even if there have been APIs connection, I do not have access to them and have not used them, so you can delete them all.

Screenshot below shows the plugin json file, is this what you were looking for?





And then the homebridge json file here:





I can use the basic auth file with username and password, cause that version were working fine.  Are you able to reset everything for me from your end? So that I can reconnect the glue lock to homebridge? 


//Taimas

**Comment by andycb** [comment]
Okay, that makes sense. I saw a few 504 errors from your homebridge plugin because your hub was too busy doing something else at the moment it tried to get the door state. These are pretty normal and can normally just be ignored, but I wonder if there's a bug in the plugin code that caused it to restart when it got this error, and regenerated the API key - which would explain why it suddenly stopped working for you. 

Just to be sure we ran a quick test against your lock and hub, and both are connected and healthy, so should respond to homebridge. 

Using an API key is probably the preferred approach, as it means you don't have your account password in a config file and should avoid this issue, so lets see if we can figure out why that didn't work for you. I've reset all your old API keys, so you should be good to generate more.

I gave it a quick test on my machine and it seemed to work okay, but I did notice that if the password contains some symbols, it can fail. You can work around this by adding quote marks like this:

```
$ curl --location --request POST 'https://user-api.gluehome.com/v1/api-keys' \
--header 'Content-Type: application/json' \
-u "example@example.com:yourPassword" \
--data-raw '{
    "name": "My Test Key",
    "scopes": ["events.read", "locks.read", "locks.write"]
}'
```

**Comment by phattrance** [comment]
Hi Andy

So I tried to create a new API key using your  workaround and I got the following error message (note, I used my email and password to the glue lock and kept the quote marks). I also tried with terminal on Mac without any results..  Are you sure that its possible to create new API keys? Cause I read on a article that clue stopped handing out API keys like a year ago ( https://glue.zendesk.com/hc/en-gb/articles/206783869-Are-there-public-APIs-for-Glue- <https://glue.zendesk.com/hc/en-gb/articles/206783869-Are-there-public-APIs-for-Glue->  )

**Comment by andycb** [comment]
Looks like the picture of the error didn't make it to GitHub. It should defiantly be possible to generate the API key - its what the homebridge plugin uses if you give it a username / password.

**Comment by phattrance** [comment]
Yes, but when I enter the command you gave to me and switched the username and password to my glue lock account) im getting the following error message:  "Please enter a correct Curl command and try again”
Similar error when I do this via terminal on Mac

So there’s defiantly something wrong with that command :)


Command i used:

$ curl --location --request POST 'https://user-api.gluehome.com/v1/api-keys <https://user-api.gluehome.com/v1/api-keys>' \
--header 'Content-Type: application/json' \
-u ”myglueemail:mygluepassword" \
--data-raw '{
    "name": "My Test Key",
    "scopes": ["events.read", "locks.read", "locks.write"]
}'

**Comment by phattrance** [comment]
Hi again, just an update that I did curl the way I did before (the way that according to you created API keys, but I couldn’t see any API keys) and here is the error message I get for it: https://i.postimg.cc/YjBLXn32/SCR-20221026-pt2-2.png <https://i.postimg.cc/YjBLXn32/SCR-20221026-pt2-2.png>

Are you able to see if there is a API key created now? If that is the case, can you just send it to me?

Cause what ever method I try to create API key here, I always get some kind of error

Thank you

**Comment by phattrance** [comment]
Hi,

So everything seems to come down to that my password is not accepted by the system? The password im using is pretty simple one, 8 characters without any special characters (its a pretty common password). So to simplify the process for us both, ive decided to change my password via the glue app, and EVEN THE APP is giving me an error while trying to change my password :D
Im not sure if you understand Swedish,  but I typed my current password, and then the new password I want and it says thats its incorrect password. So I tried to get really confused now, I tried to logout and login with the same password and that seems to be working fine, but when adding my current password to the system to change it, always got me the error thats its the incorrect password (even tho it works fine to login with). 

So i logged out if the glue app, then choose ”reset password” and added another common password and logged in via the app. I used the new password via homebridge and it seems to be accepting the username and password.

I am not able to create a new API key tho with the command you send to be before, still getting errors and I dont want to try further, incase I get this account blocked again :)

So basically it works fine now with username and password!
